### PR TITLE
Polish dev instructions

### PR DESCRIPTION
**Pull request by xiaolanz**
Disable part of pre-commit, pre-push which doesn't work.
Fix build by fixing commit version for org_pubref_rules_protobuf
Add additional instructions into dev guide.
Polish README.

### CODE REVIEW

File: bin/pre-commit [hunk 1]
```diff
@@ -30,6 +30,6 @@ BRANCH_NAME=$(git branch | grep '*' | sed 's/* //')
 if [ $BRANCH_NAME != '(no branch)' ]; then
   cd $ROOT
 
-  echo "formatting..."
-  bin/fmt.sh
+#  echo "formatting..."
```
Comment (geeknoid) [inline bin/pre-commit]: Why disable this?

File: doc/dev/development.md [hunk 2]
```diff
@@ -14,6 +14,7 @@ branch, but release branches of Broker should not change.
   - [Setting up Go](#setting-up-go)
   - [Setting up Bazel](#setting-up-bazel)
   - [Setting up Docker](#setting-up-docker)
+  - [Setting up personal access token](#setting-up-personal-access-token)
```
Comment (geeknoid) [inline doc/dev/development.md]: Once this is in here, please make the same change to the mixer's development guide.

File: doc/dev/development.md [hunk 3]
```diff
@@ -83,6 +91,7 @@ Other git workflows are also valid.
 
 1. Go to https://github.com/istio/broker
 2. Click the "Fork" button (at the top right)
+3. Follow the dialog to click your profile as the destination for the folk.
```
Comment (geeknoid) [inline doc/dev/development.md]: folk -> fork

File: doc/dev/development.md [hunk 4]
```diff
@@ -140,11 +149,18 @@ git remote set-url --push upstream no_push
 
 ### Committing changes to your fork
 
-When you're happy with some changes, you can commit them and push them to your fork:
+When you're happy with some changes, you can commit them to your fork:
```
Comment (geeknoid) [inline doc/dev/development.md]: The original wording is more precise.

You commit to your local repo, and then you push into your fork. Finally, the content will eventually be pulled from your fork into the upstream branch.

File: bin/pre-commit (hunk 1 above)
Comment (xiaolanz) [inline bin/pre-commit]: These scripts failed with this work space. some directories doesn't exist yet. If I hook the original pre-commit,  'git commit' will always fail.

File: doc/dev/development.md (hunk 4 above)
Comment (xiaolanz) [inline doc/dev/development.md]: I separate the push instructions because it requires the token.

Comment (geeknoid) [inline doc/dev/development.md]: My point is that in the sentence above, it's not correct to say "commit to your fork". You commit to your repo, and then you push to your fork.

**Pull request merged**